### DISCUSSION TITLE
New methods for Intervals

### DISCUSSION THREAD

**Discussion by Quantumplation** [aiken-lang/stdlib#65] [Ideas]
I'd like to propose adding some new methods to the standard library for comparing intervals.

```
is_entirely_after<T>(interval: Interval<T>, point: T) -> Bool // Returns true if the entire interval is after `point`
is_entirely_before<T>(interval: Interval<T>, point: T) -> Bool // Returns true if the entire interval is before `point`
```

The stdlib has `after` and `before`, but these construct half-open intervals instead of doing comparisons.

**Comment by Quantumplation** [comment]
(happy to submit a PR, just wanted to check that it was welcome first)

**Reply by KtorZ** [reply]
Sounds like something we would welcome :)

**Reply by Quantumplation** [reply]
Oh, looks like I can't yet, because we can't compare traits; I'll implement the Int version for now

**Reply by rvcas** [reply]
The pattern so far has been to accept a lambda that returns Ordering which is in the prelude

**Comment by Quantumplation** [comment]
These are merged so I'm gunna close the discussion
